packages/wrangler/src/deployment-bundle/entry.ts

```diff
@@ -55,13 +55,10 @@ export async function getEntry(
 				? path.resolve(config.site?.["entry-point"])
 				: // site.entry-point could be a directory
 					path.resolve(config.site?.["entry-point"], "index.js");
-		} else if (
-			args.legacyAssets ||
-			config.legacy_assets ||
-			args.experimentalAssets ||
-			config.experimental_assets
-		) {
+		} else if (args.legacyAssets || config.legacy_assets) {
 			file = path.resolve(getBasePath(), "templates/no-op-worker.js");
+		} else if (args.experimentalAssets || config.experimental_assets) {
```

Comment:
IMHO this should be a no-op Worker or the User Worker itself, but never the Asset Server Worker (ASW) or the Router Worker (RW). Let me explain...

The entry-point Worker will be bundled by `wrangler` via esbuild, with some very wrangler-specific esbuild configuration. In the process, `wrangler` will apply a bunch of middlewares that don't seem relevant to either ASW or RW. `wrangler`'s esbuild configuration was devised for bundling User Workers specifically, and while ASW and RW are Workers themselves, they are external to `wrangler` as a package, are self contained, and will come with their own bundling opinions and requirements. Therefore conflating bundling between the two packages, without an actual need for it, feels very wrong to me. If we want ASW and RW to be common entities to both wrangler and EWC, and to emulate production in `dev` as much as possible, ASW & RW should IMO own their own bundling + configuration, and any other devices needed for them to be deployed. To that effect, if wrangler is to have a dependency on these Workers and spin them up in a new Miniflare instance, then it should be on the already bundled versions of these Worker.

This is at least how I see the ideal world, though it remains to be seen if we can pull this off completely. But for now, this is kind of what I'm striving towards. 

Happy to hear other thoughts though ^.^

Reply:
Totally agree. I was originally thinking we'd just `import ASW from "@cloudflare/workers-shared";` from within a template and let Wrangler do it's thing, but you're completely correct — ASW2 should get it's own bundling and _that_ is what we should boot up in miniflare, with separation between this and any user worker stuff.